### PR TITLE
Validation: email test missed some cases

### PR DESCRIPTION
**Pull request by joesepi**
I found some cases that was missed by both the js regexp as well as the php regexp. @jimdelois is looking at updating the php side.

### CODE REVIEW

File: util/validate.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ define(function() {
       message: "This field must be a valid date"
     },
     Email:{
-      test: test.bind(/^[\w\.\+\-]+@(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/),
+      test: test.bind(/^[\w\.\+\-]+@(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)+([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/),
```
Comment (Aintaer) [inline util/validate.js]: Hostname portion should follow RFC for FQDNs:
```
[a-zA-Z0-9](?:[a-zA-Z0-9\-]*[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9\-]*[a-zA-Z0-9])?)*\.?
```

See
```
<hname> ::= <name>*["."<name>]
<name> ::= <letter-or-digit>[*[<letter-or-digit-or-hyphen>]<letter-or-digit>]
```

Full Regexp should be
```js
/^[\w\.\+\-]+@[a-zA-Z0-9](?:[a-zA-Z0-9\-]*[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9\-]*[a-zA-Z0-9])?)*\.?$/
```

Comment (joesepi) [inline util/validate.js]: Thanks!

**Comment by Attamusc** [comment]
:+1:

**Comment by Aintaer** [comment]
:+1:

**Pull request merged**